CMakeLists.txt

```diff
@@ -181,9 +184,9 @@ target_link_libraries(lo2s
         Nitro::dl
         Threads::Threads
         Boost::program_options
-        Boost::filesystem
         Binutils::Binutils
         fmt::fmt
+        stdc++fs
```

Comment:
That's not gone work. Have a look at https://github.com/metricq/hta 
Also g++-7 has support for filesystem.

Reply:
https://github.com/metricq/hta/blob/32c5c96b224895568372808bfb38e988e8672cb9/src/storage/file/directory.hpp#L37 if we want to support GCC 7 we would need to include this everywhere, where <filesystem> is included

Reply:
No big deal, just use the same header:
https://github.com/metricq/hta/blob/32c5c96b224895568372808bfb38e988e8672cb9/include/hta/filesystem.hpp

tbh. I have no idea why it isn't used in that file you have linked...

Reply:
Do we actually still need this choclate icecream header or can we get rid of a macro by relying on higher versions.
In any case, feel free to include  snarky comment.

Reply:
I don't see the reason to drop gcc-7 just to save on a few lines in one header...

Reply:
I don't know how relevant this is but, as far as I have looked distribution-wise, there does not really seems to be a distribution that ships with 7 that does not have a package for 8 available.

Then again I have only looked into fairly mainstream stuff, I don't know how the situation in HPC is.

Reply:
I don't know why we are discussing the support for g++-7. There are virtually no costs, just one header file. You'd need the CMake shit anyways because I bet you that soon the library will be renamed/removed.

Reply:
1. I don't even know if `experimental/filesystem` as a drop-in replacement is generally correct. And I don't want to think about it (see also  the filesystem-related proposals adopted in GCC 8 https://gcc.gnu.org/onlinedocs/libstdc++/manual/status.html#status.iso.2017)
2. Sooner or later, we're going for C++20 anyway. I would just get rid of that furuncle right now.
3. I mean including a namespace in `std`.

But if we must do it, we should at least add a
```
#pragma GCC warning "The <filesystem> header is not available, trying to use <experimental/filesystem> with a nasty workaround instead. Use at your own risk."
```